core/domain/collection_domain.py

```diff
@@ -1074,6 +1086,24 @@ def add_skill(self, skill_name):
         skill_id = CollectionSkill.get_skill_id_from_index(self.next_skill_id)
         self.skills[skill_id] = CollectionSkill(skill_id, skill_name, [])
         self.next_skill_id += 1
+        return skill_id
+
+    def get_skill_id_from_skill_name(self, skill_name):
```

Comment:
Each of the new methods added in this file needs tests (multiple tests per method covering different cases).

Reply:
Updated.